### PR TITLE
[stable13] Improve OAuth

### PR DESCRIPTION
**Pull request by rullzer**
Backport of:

- [x] https://github.com/nextcloud/server/pull/9484
- [x] https://github.com/nextcloud/server/pull/9491
- [x] https://github.com/nextcloud/server/pull/9517

### CODE REVIEW

File: lib/composer/composer/autoload_static.php [hunk 1]
```diff
@@ -6,8 +6,14 @@
 
 class ComposerStaticInit53792487c5a8370acc0b06b1a864ff4c
 {
-    public static $firstCharsPsr4 = array (
-        'O' => true,
+    public static $prefixLengthsPsr4 = array (
```
Comment (ChristophWurst) [inline lib/composer/composer/autoload_static.php]: why was this changed?

**Review by ChristophWurst — APPROVED** [review]
LGTM :+1:

File: lib/composer/composer/autoload_static.php (hunk 1 above)
Comment (rullzer) [reply lib/composer/composer/autoload_static.php]: Because I had to rerun the autoload dump. And I guess there is a new composer version ;)

**Review by MorrisJobke — APPROVED** [review]
Works 👍
